fem/gridfunc.hpp

```diff
@@ -500,49 +500,189 @@ class GridFunction : public Vector
    virtual void ProjectBdrCoefficientTangent(VectorCoefficient &vcoeff,
                                              const Array<int> &bdr_attr);
 
+   /// @brief Returns ||exsol - u_h||_L2 for scalar or vector H1 or L2 elements
+   ///
+   /// @param[in] exsol   Pointer to an array of scalar Coefficient objects,
+   ///                    one for each component of the vector field.
+   /// @param[in] irs     Optional pointer to a custom integration rule
+   ///                    e.g. higher order than the default rule.
```

Comment:
To be more precise: `exsol` and `irs` are **arrays of pointers** to `Coefficient` and `const IntegrationRule`, respectively. The size of `exsol` should be `>= vdim` with entries corresponding to the vector components of the exact solution. The size of `irs` should typically be `Geometry::NUM_GEOMETRIES`, however, only entries corresponding to geometries present in the mesh will be accessed (indexed by the `Geometry::Type` used as an index).

Reply:
Good point, `irs` is not explained properly here. I'll try to expand that as you suggest. I'll tweak the `exsol` explanation as well. You're right, it would be good to be more precise.